src/features/payment/PaymentProvider.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { createContext, useContext, useEffect, useRef, useState } from 'react';
+import type { ReactNode } from 'react';
+
+import type { AxiosError } from 'axios';
+
+import { Loader } from 'src/core/loading/Loader';
+import { useProcessNavigation } from 'src/features/instance/ProcessNavigationContext';
+import { usePaymentInformation } from 'src/features/payment/PaymentInformationProvider';
+import { PaymentStatus } from 'src/features/payment/types';
+import { usePerformPayActionMutation } from 'src/features/payment/usePerformPaymentMutation';
+import { useIsPayment } from 'src/features/payment/utils';
+import { useNavigationParam } from 'src/features/routing/AppRoutingContext';
+
+type PaymentContextProps = {
+  setLoading: (bool) => void;
+  performPayment: () => void;
+  paymentError: AxiosError | null;
+};
+
+type PaymentContextProvider = {
+  children: ReactNode;
+};
+
+export const PaymentContext = createContext<PaymentContextProps | undefined>(undefined);
+
+export const PaymentProvider: React.FC<PaymentContextProvider> = ({ children }) => {
+  const [loading, setLoading] = useState<boolean>(false);
```

Comment:
It's not clear to me why this has to be a separate state. Why not use `busy` directly, instead of setting `loading` when `busy && !paymentError`? Double state-keeping like this can lead to brittleness, as I've already seen in the `busy` state for that matter.. If this is loading should probably be a function of 'is the call to `process/next` running', which you can get from tanstack (see for example `useIsSaving()`).

Reply:
We want to also display loading when we are navigating to the external payment provider. This is not a call to process next, but to "performAction". We also want to keep displaying a loading state after the call is complete, since the redirect seems to have some slowness in displaying the page for the payment provider that is redirected to.